engine/src/flutter/impeller/display_list/canvas.cc

```diff
@@ -1776,7 +1793,8 @@ void Canvas::EndReplay() {
     BlitToOnscreen(/*is_onscreen_=*/is_onscreen_);
   }
 
-  if (!renderer_.GetContext()->FlushCommandBuffers()) {
+  if (!EnsureFinalMipmapGeneration() ||
+      !renderer_.GetContext()->FlushCommandBuffers()) {
```

Comment:
Split this into 2 different validation messages.  Muxing these will just make it harder to debug if they show up.